rustls/src/server/builder.rs

```diff
@@ -93,7 +95,11 @@ impl ConfigBuilder<ServerConfig, WantsServerCert> {
         key_der: PrivateKeyDer<'static>,
         ocsp: Vec<u8>,
     ) -> Result<ServerConfig, Error> {
-        let resolver = handy::AlwaysResolvesChain::new_with_extras(cert_chain, &key_der, ocsp)?;
+        let private_key = self
+            .state
+            .provider
+            .load_private_key(key_der)?;
+        let resolver = handy::AlwaysResolvesChain::new_with_extras(private_key, cert_chain, ocsp)?;
```

Comment:
Should we change `AlwaysResolvesChain` to have a constructor that can take a `PrivateKeyDer` and `CryptoProvider` instead?

Reply:
I think this is most general as-is (being a trivial implementation of the trait that just returns the constructor parameters). Though I think the constructor should become infallible now it does that. Will make that change.

Reply:
> Will make that change.

Done and squashed into this commit.